### PR TITLE
Implement Diff::format_email

### PR DESCRIPTION
**Pull request by Techcable**
The test is a little ugly, but at least it passes :)

### CODE REVIEW

File: libgit2-sys/lib.rs [hunk 1]
```diff
@@ -9,6 +9,8 @@ use libc::{c_char, c_int, c_uchar, c_uint, c_void, size_t};
 use libssh2_sys as libssh2;
 use std::ffi::CStr;
 
+pub const LIBGIT2_VERSION: &str = "1.0.0";
```
Comment (alexcrichton) [inline libgit2-sys/lib.rs:12]: Hm ideally this'd be auto-synthesized from libgit2 headers, would it be possible to defer this to later?

File: src/diff.rs [hunk 2]
```diff
@@ -1337,9 +1373,44 @@ impl DiffFindOptions {
...
+        } else {
+            self.raw.flags &= !opt;
+        }
+        self
+    }
+
+    /// Exclude `[PATCH]` from the subject header
+    pub fn exclude_subject_patch_header(&mut self, should_exclude: bool) -> &mut Self {
+        self.flag(raw::GIT_DIFF_FORMAT_EMAIL_EXCLUDE_SUBJECT_PATCH_MARKER, should_exclude)
+    }
```
Comment (alexcrichton) [inline src/diff.rs:1408]: Would you be interested in expanding this to include other options too?

Comment (Techcable) [reply src/diff.rs:1408]: As of right now, these are [the only flags implemented](https://github.com/libgit2/libgit2/blob/fb439c9/include/git2/diff.h#L1367-L1374)

File: libgit2-sys/lib.rs (hunk 1 above)
Comment (Techcable) [reply libgit2-sys/lib.rs]: This was only used for the test to match the version at the end. I changed it to do fuzzy matching.

**Comment by Techcable** [comment]
Thanks for the quick review! I fixed the formatting errors raised by the initial checks.

I can't seem to figure out why they're still failing.

**Comment by Techcable** [comment]
I can confirm using that patch files created with this method can be parsed by #560 (Diff::parse_buffer).
I can take an input patch file (made using libgit2), parse it, apply it, then get an identical patch file if I output it again.

There are some minor formatting differences with the patch files generated by `git format-patch`, but they don't affect patch application.

**Comment by alexcrichton** [comment]
Ok looks good to me, and thanks!

I think with a rustfmt this should be good to go.

**Pull request merged**